app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -277,6 +284,8 @@ public void onClick(View v) {
         if (!isUsingActionBar()) {
             searchbox.setText(lastQueryString);
         }
+
+        hereFunctionHandler.registerEntitySelectActivity(this);

```

Comment:
It is good practice in android to perform listener registration in `onResume` and then unregister the same listener in `onPause`. This helps prevent memory leaks due to dangling references.